
syntax = "proto3";

option java_multiple_files = true;
option java_package = "sr.grpc.gen";
option java_outer_classname = "Deliveries";

message Delivery {
  string phoneNumber = 1;
  string streetName = 2;
  int32 houseNumber = 3;
  optional int32 apartmentNumber = 4;
  message DeliveryTips {
    repeated double tip = 1;
  }
  optional DeliveryTips deliveryTips = 5;
}

message DeliveryTip {
  double deliveryAverageTips = 1;
}

service Promotion {
  rpc CountTips(Delivery) returns (DeliveryTip) {}
}
